string object_id
string direction

string surface_id
int64 x
int64 y
